### PR TITLE
Support generic identifier names with FULLTEXT INDEX syntax

### PR DESCRIPTION
**Pull request by roshan0708**
### Description

This commit adds support for different syntax of identifiers - including spaces, dots, double quotes, square brackets, etc. for `CREATE FULLTEXT INDEX` and `DROP FULLTEXT INDEX` statements.

Signed-off-by: Roshan Kanwar [rskanwar@amazon.com](mailto:rskanwar@amazon.com)

### Issues Resolved

Task: BABEL-4785

### Test Scenarios Covered ###
* **Use case based -** done

* **Boundary conditions -** done

* **Arbitrary inputs -** done

* **Negative test cases -**

* **Minor version upgrade tests -**

* **Major version upgrade tests -**

* **Performance tests -**

* **Tooling impact -**

* **Client tests -**

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: contrib/babelfishpg_tsql/src/tsqlIface.cpp [hunk 1]
```diff
@@ -6800,24 +6800,24 @@ post_process_alter_table(TSqlParser::Alter_tableContext *ctx, PLtsql_stmt_execsq
...
-    } else {
-        // No "." character found, set first to the entire string
-        table_name = table_info;
-    }
-    return std::make_pair(downcase_truncate_identifier(table_name.c_str(), table_name.length(), true),
-                           downcase_truncate_identifier(schema_name.c_str(), schema_name.length(), true));
+std::tuple<std::string, std::string, std::string> 
+getDatabaseSchemaAndTableName(TSqlParser::Table_nameContext* tctx)
+{
+	std::string table_info = ::getFullText(tctx);
```
Comment (basasairohan) [inline contrib/babelfishpg_tsql/src/tsqlIface.cpp]: This variable seems redundant. Please remove it.

File: contrib/babelfishpg_tsql/src/tsqlIface.cpp [hunk 2]
```diff
@@ -6800,24 +6800,24 @@ post_process_alter_table(TSqlParser::Alter_tableContext *ctx, PLtsql_stmt_execsq
...
+
+	if (tctx->database)
+		db_name = stripQuoteFromId(tctx->database);
+	if (tctx->schema)
+		schema_name = stripQuoteFromId(tctx->schema);
+	if (tctx->table)
+		table_name = stripQuoteFromId(tctx->table);
+
+	return std::make_tuple(downcase_truncate_identifier(db_name.c_str(), db_name.length(), true),
+							downcase_truncate_identifier(schema_name.c_str(), schema_name.length(), true),
```
Comment (basasairohan) [inline contrib/babelfishpg_tsql/src/tsqlIface.cpp]: nit : indentation

**Review by basasairohan — CHANGES_REQUESTED** [review]

File: test/JDBC/input/full_text_search/BABEL-4785-vu-cleanup.sql [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+GO
+
+DROP SCHEMA IF EXISTS [fts_test  .schema];
+GO
+
+USE master;
+GO
+
+DROP DATABASE IF EXISTS "fts_test .db";
+GO
```
Comment (rishabhtanwar29) [inline test/JDBC/input/full_text_search/BABEL-4785-vu-cleanup.sql]: Nit: add newline at end of each new file.

**Review by rishabhtanwar29 — APPROVED** [review]
Let's have a generic title like: 
Support generic identifier names with FULLTEXT INDEX syntax

File: test/JDBC/expected/BABEL-4785-vu-prepare.out [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+CREATE TABLE fts_schema_test.[table.test](id int not null, name text)
+GO
+
+CREATE UNIQUE INDEX fti_schema_test8 ON fts_schema_test.[table.test](id);
+GO
+
+CREATE FULLTEXT INDEX ON fts_schema_test.[table.test](name) KEY INDEX fti_schema_test8;
+GO
+
+CREATE TABLE fts_schema_test."fts object with dots and spaces in quotes"(id int not null, name text)
```
Comment (rishabhtanwar29) [inline test/JDBC/expected/BABEL-4785-vu-prepare.out:74]: Let's have test with special characters as well like {}\/ # etc

**Review by basasairohan — APPROVED** [review]
LGTM

**Review by shardgupta — APPROVED** [review]

**Pull request merged**